src/colvarproxy.cpp

```diff
@@ -737,3 +665,92 @@ int colvarproxy::get_version_from_string(char const *version_string)
   is >> newint;
   return newint;
 }
+
+
+void colvarproxy::smp_stream_error()
+{
+  cvm::error("Error: trying to access an output stream from a "
+             "multi-threaded region (bug).  For a quick workaround, use "
+             "\"smp off\" in the Colvars config.\n", BUG_ERROR);
+}
+
+
+std::ostream * colvarproxy::output_stream(std::string const &output_name,
+                                          std::ios_base::openmode mode)
+{
+  if (cvm::debug()) {
+    cvm::log("Using colvarproxy::output_stream()\n");
+  }
+
+  std::ostream *os = get_output_stream(output_name);
+  if (os != NULL) return os;
+
+  if (!(mode & (std::ios_base::app | std::ios_base::ate))) {
+    backup_file(output_name);
+  }
+  std::ofstream *osf = new std::ofstream(output_name.c_str(), mode);
+  if (!osf->is_open()) {
+    cvm::error("Error: cannot write to file/channel \""+output_name+"\".\n",
+               FILE_ERROR);
+    return NULL;
+  }
+  output_stream_names.push_back(output_name);
+  output_files.push_back(osf);
+  return osf;
+}
+
+
+std::ostream *colvarproxy::get_output_stream(std::string const &output_name)
+{
+  if (smp_enabled() == COLVARS_OK) {
```

Comment:
I can't get over this function returning the Boolean opposite of what's expected from its name... I see the point in the distinction between disabled and not implemented, but it's not used in the current code.

Reply:
I am fine if you want to replace this function with two functions, each of which returns a `bool`. Please go ahead.